assets/src/stories-editor/components/block-mover/draggable.js

```diff
@@ -75,8 +89,81 @@ class Draggable extends Component {
 	 *
 	 * @param  {Object} event The non-custom DragEvent.
 	 */
-	onDragOver = ( event ) => {
+	onDragOver = ( event ) => { // eslint-disable-line complexity
+		const {
+			blockName,
+			snapLines,
+			clearSnapLines,
+			setSnapLines,
+			parentBlockElement,
+			horizontalSnaps,
+			verticalSnaps,
+		} = this.props;
+
 		const top = parseInt( this.cloneWrapper.style.top ) + event.clientY - this.cursorTop;
+		const left = parseInt( this.cloneWrapper.style.left ) + event.clientX - this.cursorLeft;
+
+		if ( top === lastY && left === lastX ) {
+			return;
+		}
+
+		const isCTABlock = 'amp/amp-story-cta' === blockName;
+
+		// Get the correct dimensions in case the block is rotated, as rotation is only applied to the clone's inner element(s).
+		// For CTA blocks, not the whole block is draggable, but only the button within.
+		const blockElement = isCTABlock ? this.cloneWrapper.querySelector( '.amp-story-cta-button' ) : this.cloneWrapper.querySelector( '.wp-block' );
+
+		// We calculate with the block's actual dimensions relative to the page it's on.
+		const {
+			top: actualTop,
+			right: actualRight,
+			bottom: actualBottom,
+			left: actualLeft,
+		} = getRelativeElementPosition( blockElement, parentBlockElement );
+
+		const horizontalCenter = ( actualRight + actualLeft ) / 2;
+		const verticalCenter = ( actualTop + actualBottom ) / 2;
+
+		const newSnapLines = [];
+
+		const snappingEnabled = ! event.getModifierState( 'Alt' );
+
+		if ( snappingEnabled ) {
+			// Go through all snap targets and find the one that is closest.
+			const findSnap = ( snapKeys, ...values ) => {
+				return values
+					.map( ( value ) => {
+						const snap = findClosestSnap( value, snapKeys, BLOCK_DRAGGING_SNAP_GAP );
+						return [ value, snap ];
+					} )
+					.filter( ( arr ) => arr[ 1 ] !== null )
```

Comment:
What does this filter do again? When would it be `null`?

Reply:
Ah of course, if there is no closest snap to the given value, it will be null.

Reply:
`arr[ 1 ]` is the `snap` variable from the `[ value, snap ]` array above, i.e. the result of `findClosestSnap`. It is `null` when no snap target was found for the given `value`.

The thinking here was that I need a map of value -> snap target to find the most suitable snap target. However, when I look at this code again now it doesn't even use `arr[ 0 ]` 🤔